sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isReadOnly() {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getSeparator() {
+        return "/";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Iterable<Path> getRootDirectories() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Iterable<FileStore> getFileStores() {
+        return
+            this.fileStores.values();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Set<String> supportedFileAttributeViews() {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Path getPath(String s, String... strings) {
+        return new AzurePath(this, s, strings);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PathMatcher getPathMatcher(String s) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public UserPrincipalLookupService getUserPrincipalLookupService() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public WatchService newWatchService() throws IOException {
+        throw new UnsupportedOperationException();
+    }
+
+    String getFileSystemName() {
+        return this.blobServiceClient.getAccountName();
+    }
+
+    BlobServiceClient getBlobServiceClient() {
+        return this.blobServiceClient;
+    }
+
+    private BlobServiceClient buildBlobServiceClient(String accountName, Map<String,?> config) {
+        // Build the endpoint.
+        String scheme = !config.containsKey(AZURE_STORAGE_USE_HTTPS)
+                || (Boolean) config.get(AZURE_STORAGE_USE_HTTPS)
+                ? "https" : "http";
+        BlobServiceClientBuilder builder = new BlobServiceClientBuilder()
+                .endpoint(String.format(AZURE_STORAGE_ENDPOINT_TEMPLATE, scheme, accountName));
+
+        // Set the credentials.
+        if (config.containsKey(AZURE_STORAGE_ACCOUNT_KEY)) {
+            builder.credential(new StorageSharedKeyCredential(accountName,
+                    (String)config.get(AZURE_STORAGE_ACCOUNT_KEY)));
+        }
+        else if (config.containsKey(AZURE_STORAGE_SAS_TOKEN)) {
+            builder.sasToken((String) config.get(AZURE_STORAGE_SAS_TOKEN));
+        }
+        else {
+            throw new IllegalArgumentException(String.format("No credentials were provided. Please specify one of the" +
+                    " following when constructing an AzureFileSystem: %s, %s.", AZURE_STORAGE_ACCOUNT_KEY,
+                    AZURE_STORAGE_SAS_TOKEN));
+        }
+
+        // Configure options and client.
+        builder.httpLogOptions(new HttpLogOptions()
+            .setLogLevel((HttpLogDetailLevel)config.get(AZURE_STORAGE_HTTP_LOG_DETAIL_LEVEL)));
```

Comment:
This should use `BlobServiceClientBuilder.getDefaultHttpLopOptions()` as the base before setting the log level, this will continue to maintain proper query string and header logging. Right now this would wipe out all Storage specific headers and query string parameters from being logged.

Future enhancement would be adding two additional configurations that would set the loggable query string parameters and header names.

Reply:
Thanks for catching this. 

Yea I felt like those parameters were a bit overkill for an initial release and that it would be easy to add them later.

Reply:
Agreed on the parameters not being needed during the initial preview.